### PR TITLE
Fix error handling in `LdapOrgReaderProcessor`, and support complex paging options

### PR DESCRIPTION
**Pull request by freben**
The paging options were documented and supported in the types - but the config reading code was broken and assumed only booleans.

### CODE REVIEW

File: plugins/catalog-backend/src/ingestion/processors/ldap/client.ts [hunk 1]
```diff
@@ -31,8 +32,20 @@ import {
...
+    target: string,
+    bind?: BindConfig,
+  ): Promise<LdapClient> {
     const client = ldap.createClient({ url: target });
+
+    // We want to have a catch-all error handler at the top, since the default
+    // behavior of the client is to blow up the entire process when it fails,
+    // unless an error handler is set.
+    client.on('error', (err: ldap.Error) => {
+      logger.warn(`LDAP client threw an error, ${errorString(err)}`);
```
Comment (Rugvip) [inline plugins/catalog-backend/src/ingestion/processors/ldap/client.ts:46]: Prolly better to log this with error level right?

File: plugins/catalog-backend/src/ingestion/processors/ldap/client.ts [hunk 2]
```diff
@@ -62,6 +75,10 @@ export class LdapClient {
       return await new Promise<SearchEntry[]>((resolve, reject) => {
         const output: SearchEntry[] = [];
 
+        this.client.on('error', (err: ldap.Error) => {
```
Comment (Rugvip) [inline plugins/catalog-backend/src/ingestion/processors/ldap/client.ts:78]: There's no guarantee that this error has anything to do with the existing query right? Might be better to skip this? I'm assuming the `search` callback always gets called, and the error passed in there is the one to use.

**Review by Rugvip — COMMENTED** [review]
👍 , just a few thoughts

File: plugins/catalog-backend/src/ingestion/processors/ldap/client.ts (hunk 2 above)
Comment (freben) [reply plugins/catalog-backend/src/ingestion/processors/ldap/client.ts:78]: Let's remove this one actually

File: plugins/catalog-backend/src/ingestion/processors/ldap/client.ts (hunk 1 above)
Comment (freben) [reply plugins/catalog-backend/src/ingestion/processors/ldap/client.ts:46]: K!

**Pull request merged**